File: lift-impact-platform/frontend/services/api_client.py
from __future__ import annotations

import requests

BASE_URL = "http://localhost:8000"


def upload(file_bytes: bytes, filename: str) -> dict:
    r = requests.post(f"{BASE_URL}/upload", files={"file": (filename, file_bytes)})
    r.raise_for_status()
    return r.json()


def get_checklist(file_id: str) -> dict:
    r = requests.get(f"{BASE_URL}/checklist/{file_id}")
    r.raise_for_status()
    return r.json()


def get_eda(file_id: str, metric_group: str = "Suggestions", variable: str | None = None, include_zscore: bool = False) -> dict:
    params = {"metric_group": metric_group, "include_zscore": include_zscore}
    if variable:
        params["variable"] = variable
    r = requests.get(f"{BASE_URL}/eda/{file_id}", params=params)
    r.raise_for_status()
    return r.json()


def run_model(file_id: str, scenario_multiplier: float = 1.0, isolate_channel: str | None = None) -> dict:
    r = requests.post(
        f"{BASE_URL}/run/{file_id}",
        json={"scenario_multiplier": scenario_multiplier, "isolate_channel": isolate_channel},
    )
    r.raise_for_status()
    return r.json()

File: lift-impact-platform/frontend/src/services/apiClient.js
import axios from "axios";

const BASE_URL = import.meta.env.VITE_API_BASE_URL || "http://localhost:8000";

const api = axios.create({
  baseURL: BASE_URL,
  timeout: 30000,
});

const extractError = (error) => {
  if (error.code === "ECONNABORTED") {
    return `Request timed out while connecting to ${BASE_URL}`;
  }
  if (error.response) {
    return `API error ${error.response.status}: ${JSON.stringify(error.response.data)}`;
  }
  if (error.request) {
    return `Could not connect to backend at ${BASE_URL}. Start API server with: uvicorn backend.api.main:app --reload --port 8000`;
  }
  return error.message || "Unexpected frontend request error";
};

export async function uploadExcel(file) {
  const form = new FormData();
  form.append("file", file, file.name);
  try {
    const response = await api.post("/upload", form, {
      headers: { "Content-Type": "multipart/form-data" },
    });
    return response.data;
  } catch (error) {
    throw new Error(extractError(error));
  }
}

export async function fetchEda(fileId) {
  try {
    const response = await api.get(`/eda/${fileId}`);
    return response.data;
  } catch (error) {
    throw new Error(extractError(error));
  }
}

export async function runModel(fileId, scenarioMultiplier, isolateChannel) {
  try {
    const response = await api.post(`/run/${fileId}`, {
      scenario_multiplier: Number(scenarioMultiplier),
      isolate_channel: isolateChannel || null,
    });
    return response.data;
  } catch (error) {
    throw new Error(extractError(error));
  }
}
